src/XMakeBuildEngine/Definition/Project.cs

```diff
@@ -1032,6 +1033,170 @@ public static string GetEvaluatedItemIncludeEscaped(ProjectItemDefinition item)
+                }
+                else
+                {
+                    var occurrencesInInclude = ItemMatchesInSpecCompareViaExpander(itemValue, itemElement.Include, (s, o) => expander.ExpandIntoStringLeaveEscaped(s, o, itemElement.IncludeLocation), out provenance);
+
+                    if (occurrencesInInclude > 0)
+                    {
+                        result.Add(new ProvenanceResult(itemElement, Operation.Include, provenance, occurrencesInInclude));
+                    }
+                }
+            }
+
+            return result;
+        }
+
+        /// <summary>
+        /// Since:
+        ///     - we have no proper AST and interpreter for itemspecs that we can do analysis on
+        ///     - GetItemProvenance needs to have correct counts for exclude strings (as correct as it can get while doing it after evaluation)
+        /// 
+        /// The temporary hack is to use the expander to expand the strings, and if any property or item references were encountered, return Provenance.Inconclusive
+        /// </summary>
+        /// <param name="itemToMatch"></param>
+        /// <param name="itemSpec"></param>
+        /// <param name="expand"></param>
+        /// <param name="provenance"></param>
+        /// <returns></returns>
+        private int ItemMatchesInSpecCompareViaExpander(string itemToMatch, string itemSpec, Func<string, ExpanderOptions, string> expand, out Provenance provenance)
+        {
+            if (string.IsNullOrEmpty(itemSpec))
+            {
+                provenance = Provenance.Undefined;
+                return 0;
+            }
+
+            // look into the itemspec as if it were expanded by the Expander
+            Provenance provenanceFromExpandedPropertiesAndItems;
+            var expandedMatches = ItemMatchesInSpec(itemToMatch, expand(itemSpec, ExpanderOptions.ExpandPropertiesAndItems), out provenanceFromExpandedPropertiesAndItems);
+
+            // look into the itemspec when properties are expanded, but not items
+            Provenance provenanceFromExpandedProperties;
+            ItemMatchesInSpec(itemToMatch, expand(itemSpec, ExpanderOptions.ExpandProperties), out provenanceFromExpandedProperties);
+
+            // look into the raw itemspec
+            Provenance provenanceFromNonExpandedString;
+            var nonExpandedMatches = ItemMatchesInSpec(itemToMatch, itemSpec, out provenanceFromNonExpandedString);
+
+            if (expandedMatches > nonExpandedMatches)
+            {
+                // return the number of occurences when properties AND items are expanded to get the correct occurence count
+
+                // return the provenance WITHOUT item expansion. Otherwise the items coming from a referenced item get interpreted as StringLiteral
+                // include="*.cs;@(Compile)" needs to return Inconclusive|Glob and not Inconclusive|Glob|StringLiteral
```

Comment:
If I match `itemToMatch` against the expanded string with `ExpanderOptions.ExpandPropertiesAndItems`, then I will get bad data from items coming inside an item reference.

For example, computing GetItemProvenance("Program.cs"):
```
a include="*.cs"
b include="@(a)"
```
For `a`, I want to get `Provenance.FromGlob` and Occurrences=1
For `b` I want to get `Provenance.Inconclusive` and Occurrences=1

However, for `b`, if I expand items, I will get `Provenance.Inconclusive | Provenance.FromLiteral`. This is because the expander loads the items for `a` from its already evaluated items, so all the globs will have been expanded and I only get string literals.

Ideally, for `b`, I'd want GetItemProvenance to return actual nodes from some AST / interpretation dag that matched the string. But until we have such a structure, I am going to not flow information from indirect item references and add  `Inconclusive` to the bit flag. This is accomplished by matching against a [Property only expansion](https://github.com/Microsoft/msbuild/pull/809/files#diff-fc487db8bc5bcfe97c6c0996f8b14fefR1114).

For properties though, the expander gives them as is (does not expand globs, or follow item reference). So for properties, I am actually providing 'correct' provenance for literals and globs, but I still add `Inconclusive`, to signify that a reference has been traversed to get that information.